python_fixturify_project/project.py

```diff
@@ -97,11 +82,47 @@ def write(self, dir_json):
         """Writes the existing files property containing a directory representation to the tmp directory."""
         if dir_json:
             self.merge_files(dir_json)
-        self.write_project()
+        self.__write_project()
+
+    def read(self):
+        """Reads the contents of the base_dir to a dict"""
+        files: Dict[str, Any] = {}
+
+        for path in glob.iglob(self.base_dir + "**/**", recursive=True):
+            rel_path = os.path.relpath(path, self.base_dir)
 
-    def write_project(self):
+            t = "File" if os.path.isfile(path) else "Dir"
+            name = os.path.basename(path)
+            p = os.path.dirname(path)
+
+            if rel_path == ".":
+                continue
+
+            if os.path.isfile(path):
+                with open(path) as f:
+                    self.__add_file(files, rel_path, f.read())
+            else:
+                self.__add_dir(files, rel_path)
+
+        return files
+
+    def __add_file(self, files, path, contents):
+        file = os.path.basename(path)
+        dir_name = os.path.dirname(path)
+
+        if dir_name != "":
+            self.__add_dir(files, dir_name)[file] = contents
+        else:
+            files[file] = contents
```

Comment:
What happens if we need files with the same name, just in different directories? Will this overwrite the contents with the last file read that shares the same name?

Reply:
This is only in the case of the top level, where we want to avoid the song and dance of determining the dir structure. You can't have multiple files with the same name, so this should be 'safe'.

Reply:
I can add a test that verifies this.

Reply:
Sounds good, thank you!

Reply:
Added a test to demonstrate this is 'safe'.

Reply:
Nice!